akka-actor/src/main/scala/akka/event/DeadLetterListener.scala

```diff
@@ -4,15 +4,18 @@
 
 package akka.event
 
-import akka.actor.Actor
-import akka.actor.DeadLetter
+import akka.actor.{ Actor, Cancellable, DeadLetter }
 import akka.event.Logging.Info
 
+import scala.concurrent.duration.FiniteDuration
+
 class DeadLetterListener extends Actor {
 
-  val eventStream = context.system.eventStream
-  val maxCount = context.system.settings.LogDeadLetters
-  var count = 0
+  private val eventStream = context.system.eventStream
+  private val maxCount = context.system.settings.LogDeadLetters
+  private val suspendDuration = context.system.settings.LogDeadLettersSuspendDuration
+  private var count = 0
+  private var reEnableOngoing: Cancellable = _
```

Comment:
I wonder if it wouldn't it be simpler to store the 'deadline' in the `var` instead of scheduling?

Reply:
Yes, That's more simpler, and don't need a message.